### PR TITLE
Rename `data_wp_context` to `wp_interactivity_data_wp_context`

### PR DESCRIPTION
**Pull request by swissspidy**
Renames the function for more clarity.

**Important:** requires a corresponding change to the block render callbacks in Gutenberg

Trac ticket: https://core.trac.wordpress.org/ticket/60575

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by swissspidy** [comment]
cc @luisherranz

**Comment by gziolo** [comment]
It works for me. @c4rl0sbr4v0 and @darerodz, can we proceed with it? It will require changes in WordPress packages for core blocks and adjustment to the dev note.

File: tests/phpunit/tests/interactivity-api/interactivity-api.php [hunk 1]
```diff
@@ -314,18 +314,18 @@ public function test_process_directives_only_process_the_root_interactive_blocks
 	}
 
 	/**
-	 * Tests that data_wp_context function correctly converts different array
+	 * Tests that wp_interactivity_wp_interactivity_data_wp_context function correctly converts different array
```
Comment (gziolo) [inline tests/phpunit/tests/interactivity-api/interactivity-api.php]:
```suggestion
	 * Tests that wp_interactivity_data_wp_context function correctly converts different array
```

File: tests/phpunit/tests/interactivity-api/interactivity-api.php [hunk 2]
```diff
@@ -314,18 +314,18 @@ public function test_process_directives_only_process_the_root_interactive_blocks
...
 	 * structures to a JSON string.
 	 *
 	 * @ticket 60356
 	 *
-	 * @covers ::data_wp_context
+	 * @covers ::wp_interactivity_wp_interactivity_data_wp_context
```
Comment (gziolo) [inline tests/phpunit/tests/interactivity-api/interactivity-api.php]:
```suggestion
	 * @covers ::wp_interactivity_data_wp_context
```

**Review by gziolo — APPROVED** [review]

File: tests/phpunit/tests/interactivity-api/interactivity-api.php (hunk 1 above)
Comment (swissspidy) [reply tests/phpunit/tests/interactivity-api/interactivity-api.php]: Double trouble 😄

**Comment by c4rl0sbr4v0** [comment]
Gutenberg version:
https://github.com/WordPress/gutenberg/pull/59465

**Comment by cbravobernal** [comment]
> It works for me. @c4rl0sbr4v0 and @DAreRodz, can we proceed with it? It will require changes in WordPress packages for core blocks and adjustment to the dev note.

Sure @gziolo .
Dev note is updated. Handbook and Gutenberg updates are available here:

https://github.com/WordPress/gutenberg/pull/59465

**Comment by swissspidy** [comment]
Thanks y'all! Committed in https://core.trac.wordpress.org/changeset/57742

**Pull request closed without merging**